extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java

```diff
@@ -166,7 +167,7 @@ public ListenableFuture<TaskStatus> run(Task task)
                     PeonCommandContext context = new PeonCommandContext(
                         generateCommand(task),
                         javaOpts(task),
-                        dirTracker.getTaskDir(task.getId()),
+                        new File(taskConfig.getBaseTaskDirPath()),
```

Comment:
`getBaseTaskDirPath` is for JSON backwards compat; it may be null and isn't used elsewhere. Better to do `getBaseTaskDirPaths().get(0)`. Please also add `@Deprecated` to `TaskConfig.getBaseTaskDirPath` to make it clearer to others that this method shouldn't be used.